edb/edgeql/compiler/stmtctx.py

```diff
@@ -214,10 +214,15 @@ def fini_expression(
 
     ctx.path_scope.validate_unique_ids()
 
+    # Collect query parameters
+    params = collect_params(ctx)
+
     # ConfigSet and ConfigReset don't like being part of a Set, so bail early
     if isinstance(ir.expr, (irast.ConfigSet, irast.ConfigReset)):
         ir.expr.scope_tree = ctx.path_scope
         ir.expr.globals = list(ctx.env.query_globals.values())
+        ir.expr.params = params
+        ir.expr.schema = ctx.env.schema
```

Comment:
@msullivan from the top of your head, is there an obvious blocker that prevents ConfigCommand be a part of Statement?

If you don't think of one, I'll try to fit it in (in a following PR)